star_frame/src/unsize/impls/map.rs

```diff
@@ -148,6 +148,14 @@ where
         }
     }
 
+    /// Inserts all key-value pairs from the provided iterator into the map.
+    #[exclusive]
+    pub fn insert_all(&mut self, values: impl IntoIterator<Item = (K, V)>) -> Result<()> {
+        values
+            .into_iter()
+            .try_for_each(|(key, value)| self.insert(key, value).map(|_| ()))
+    }
```

Comment:
Is it worth returning how many were inserted or something like that? I guess that can be determined from just checking len before and after.